File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.java
class Solution {
    public int evalRPN(String[] tokens) {
        List<String> ops = Arrays.asList("+","-","/","*");
        Stack<String> stk = new Stack<>();
        for (String ch:tokens){
            System.out.println(stk);
            if (!ops.contains(ch)){
                stk.push(ch);
            }else{
                int b = Integer.parseInt(stk.pop());
                int a = Integer.parseInt(stk.pop());
                if (ch.equals("+")){
                    stk.push(Integer.toString(a+b));
                }else if(ch.equals("-")){
                    stk.push(Integer.toString(a-b));
                }else if(ch.equals("*")){
                    stk.push(Integer.toString(a*b));
                }else if(ch.equals("/")){
                    stk.push(Integer.toString(a/b));
                }
            }
        }
        return Integer.parseInt(stk.pop());
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        prens = {")":"(", "]":"[", "}":"{"}
        stk = []
        for i in s:
            if i not in prens.keys():
                stk.append(i)
                # print(stk)
            else:
                if not stk:
                    return False
                else:
                    if stk[-1]==prens[i]:
                        stk.pop()
                    else:
                        return False
        if not stk:
            return True
        return False